### PR TITLE
Add friends' permissions mechanism

### PR DESCRIPTION
**Pull request by OfirMatasas**
## Description: 
Adding the ability to control over the permissions of every friend for creating notifications.
If the friend has the permission to do so, the notification's status will be initialized to "Active". If not - "Pending".

## Type of change:
- [ ] Bug fix
- [x] New feature
- [ ] Fix/new infrastructure

## Background context of the changes:
The user should have the option to choose which friends can automatically create location, time and/or dynamic notifications without waiting for being approved by the user.

## The ticket this PR closes:
[NOTIFY-182](https://ofirlindekel.atlassian.net/browse/NOTIFY-182)

## Scenarios that were tested during this implementation:
- The friend have permission to create a notification for the user (any type).
- The friend doesn't have permission to automatically create a notification for the user. 

## Insert screenshots / video of test results (for all relevant platforms)
https://github.com/OfirMatasas/Notify/assets/88926592/6597819d-1e29-40b9-acae-c531394c0bb0

### CODE REVIEW

File: Notify/Notify.Functions/Notify.Functions/NotifyFunctions/Friend/DeleteFriend.cs [hunk 1]
```diff
@@ -46,7 +42,7 @@ public static class DeleteFriend
             }
             else
             {
-                message = $"Friendship between {username} and {friendName} was deleted";
+                message = $"Friendship and permissions between {username} and {friendName} was deleted";
```
Comment (LinMoskovitch) [inline Notify/Notify.Functions/Notify.Functions/NotifyFunctions/Friend/DeleteFriend.cs:45]:
```suggestion
                message = $"Friendship and permissions between {username} and {friendName} were deleted";
```

File: Notify/Notify/Notify/ViewModels/FriendDetailsPageViewModel.cs [hunk 2]
```diff
@@ -13,21 +18,77 @@ public class FriendDetailsPageViewModel : INotifyPropertyChanged
         public string UserName { get; set; }
         public string Telephone { get; set; }
         public ImageSource ProfileImage { get; set; }
-
+        public List<string> AllowDisallowOptions { get; set; } = new List<string> { Constants.NOTIFICATION_PERMISSION_ALLOW, Constants.NOTIFICATION_PERMISSION_DISALLOW };
```
Comment (LinMoskovitch) [inline Notify/Notify/Notify/ViewModels/FriendDetailsPageViewModel.cs:21]: Consider changing the name to something like: `PermissionOptions `

File: Notify/Notify/Notify/ViewModels/NotificationDetailsPageViewModel.cs [hunk 3]
```diff
@@ -117,6 +120,42 @@ private async void onRenewNotificationButtonClicked()
                     "OK");
             }
         }
+
```
[the remark this answers is not in the document]
Comment (OfirMatasas) [reply Notify/Notify/Notify/ViewModels/NotificationDetailsPageViewModel.cs:123]: Discussed.

**Review by LinMoskovitch — APPROVED** [review]
LGTM

**Pull request merged**